src/winner.js

```diff
@@ -0,0 +1,19 @@
+export default function calculateWinner(squares) {
+    const lines = [
+      [0, 1, 2],
+      [3, 4, 5],
+      [6, 7, 8],
+      [0, 3, 6],
+      [1, 4, 7],
+      [2, 5, 8],
+      [0, 4, 8],
+      [2, 4, 6],
+    ];
+    for (let i = 0; i < lines.length; i++) {
+      const [a, b, c] = lines[i];
+      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
+        return squares[a];
+      }
```

Comment:
This logic is ok, but as a nice thought exercise you may want to explore a more functional way to approach it that would also be flexible to larger matrix boards (4x4, 5x5)

Reply:
I like that 👍